include/Target/Btor/BtorToBtorIRTranslation.h

```diff
@@ -0,0 +1,170 @@
+        modelFile = fopen(s.c_str(), "r");
+    }
+
+  ~Deserialize() {
+      if (model) {
+        btor2parser_delete(model);
+      }
+      if (modelFile) {
+        fclose(modelFile);
+      }
+  }
+
+///===----------------------------------------------------------------------===//
+/// Parse btor2 file
+///===----------------------------------------------------------------------===//
+  
+  bool parseModelIsSuccessful();
+
+///===----------------------------------------------------------------------===//
+/// Create MLIR module
+///===----------------------------------------------------------------------===//
+  
+  std::map<int64_t, Value> cache;
+
+  OwningOpRef<FuncOp> buildInitFunction();
+  OwningOpRef<FuncOp> buildNextFunction();
+  
+ private: 
+///===----------------------------------------------------------------------===//
+/// Parse btor2 file
+///===----------------------------------------------------------------------===//
+  
+  Btor2Parser *model = nullptr;
+  FILE *modelFile = nullptr;
+
+  std::vector<Btor2Line *> inputs;
+  std::vector<Btor2Line *> states;
+  std::vector<Btor2Line *> bads;
+  std::vector<Btor2Line *> inits;
+  std::vector<Btor2Line *> nexts;
+  std::vector<Btor2Line *> constraints;
+ 
+  std::map<int64_t, Btor2Line *> reachedLines;
+
+  void parseModelLine(Btor2Line *l);
+
+///===----------------------------------------------------------------------===//
+/// Create MLIR module
+///===----------------------------------------------------------------------===//
+
+  MLIRContext *context;
+  OpBuilder builder;
+  Location unknownLoc;
+  
+  void toOp(Btor2Line *line);
+  bool isValidChild(Btor2Line * line);
+  void createNegateLine(int64_t curAt, Value &child);
+  Operation * createMLIR(const Btor2Line *line, const int64_t *kids);
+
+  template <typename btorOp>
+  Operation * buildBinaryOp(Value &lhs, Value &rhs) {
+    auto res = builder.create<btorOp>(unknownLoc, lhs, rhs);
+    return res;
+  }
+
+  template <typename btorOp>
+  Operation * buildComparisonOp(btor::BtorPredicate pred,
+                                Value &lhs, Value &rhs) {
+    auto res = builder.create<btorOp>(unknownLoc, pred, lhs, rhs);
+    return res;
+  }
+
+  template <typename btorOp>
+  Operation * buildOverflowOp(Value &lhs, Value &rhs) {
+    auto res = builder.create<btorOp>(unknownLoc, 
+                                    builder.getIntegerType(1), 
+                                    lhs, rhs);
+    return res;
+  }
+
+  Operation * buildConstantOp(uint32_t width, const std::string &str, uint32_t radix) {
+    Type type = builder.getIntegerType(width);
+    mlir::APInt value(width, 0, radix);
+    if (str.compare("ones") == 0) {
+      value.setAllBits();
+    } else if (str.compare("one") == 0) {
+      value = mlir::APInt(width, 1, radix);
+    } else if (str.compare("zero") != 0) {
+      value = mlir::APInt(width, str, radix);
+    }
+    auto res = builder.create<btor::ConstantOp>(unknownLoc, type,
```

Comment:
should warn if the constant does not fit into `getLimitedValue`. 
More generally, the attribute should be arbitrary precision and not a fixed size integer.